src/ucs/stats/stats.c

```diff
@@ -460,6 +477,184 @@ void ucs_stats_node_free(ucs_stats_node_t *node)
     }
 }
 
+static uint32_t
+ucs_stats_clsid_to_agrgt_sum_base_offset_get(ucs_stats_node_t *node,
+                                             size_t num_counters_in_class)
+{
+    uint32_t offset                      = 0;
+    ucs_stats_filter_node_t *filter_node = node->filter_node;
+    const char *class_name               = node->cls->name;
+    unsigned clsid                       = node->cls->class_id;
+    unsigned *class_id                   = &node->cls->class_id;
+    ucs_stats_node_t *temp_node;
+    uint32_t counter_index;
+    uint32_t cnt_agrgt_sum_index;
+
+    pthread_mutex_lock(&ucs_stats_context.lock);
+
+    if (ucs_likely(clsid != 0) && (ucs_stats_context.aggrgt_sum_db)) {
+        offset = ucs_stats_context.aggrgt_sum_db[clsid]
+                         .aggregate_sum_class_offset;
+    } else {
+        size_t total_alloc_size_bytes;
+
+        if (ucs_stats_context.aggrgt_sum_db_size >=
+            ucs_stats_context.aggrgt_sum_db_alloc_size) {
+            ucs_stats_context.aggrgt_sum_db_alloc_size +=
+                    UCS_STATS_AGG_SUM_COUNTERS_REALLOC_SZ;
+
+            /* Reallocate clsid->name db */
+            total_alloc_size_bytes =
+                    (ucs_stats_context.aggrgt_sum_db_alloc_size *
+                     sizeof(*ucs_stats_context.aggrgt_sum_db));
+            ucs_stats_context.aggrgt_sum_db = ucs_realloc(
+                    ucs_stats_context.aggrgt_sum_db, total_alloc_size_bytes,
+                    (const char*)UCS_STATS_CLSID_TO_NAME_DB_NAME);
+        }
+
+        /* Initialize entry */
+        ucs_snprintf_safe(
+                ucs_stats_context
+                        .aggrgt_sum_db[ucs_stats_context.aggrgt_sum_db_size]
+                        .class_name,
+                sizeof(ucs_stats_context
+                               .aggrgt_sum_db[ucs_stats_context.aggrgt_sum_db_size]
+                               .class_name),
+                "%s", class_name);
```

Comment:
where is ucs_stats_aggrgt_sum_clsid_to_name_db_t.class_name field is used?

Reply:
OK, you are right, it's indeed not in use. Removed it.